lib/Controller/PreferencesController.php

```diff
@@ -74,7 +80,11 @@ public function update($key, $value): JSONResponse {
 			throw new ClientException('key or value missing');
 		}
 
-		$newValue = $this->userPreference->setPreference($key, $value);
+		try {
+			$newValue = $this->userPreference->setPreference($this->userId, $key, $value);
+		} catch (PreConditionNotMetException $e) {
+			throw new ClientException('key or value invalid');
```

Comment:
Please remove try-catch here. The method will not throw a PreConditionNotMetException for us.